### DISCUSSION TITLE
How to navigate to the fixture definition in Visual Studio Code?

### DISCUSSION THREAD

**Discussion by jonathanmach** [pytest-dev/pytest#9182] [Q&A] [answered]
Hi all, I've been searching all over the internet for an answer to this question but I haven't found anything yet. 
When 
For example, in the following test, how to quickly go to the file which my `test_blogpost` fixture is defined so that I can examine its content?

```python
    def test_not_authorized_to_publish(self, tapp, test_blogpost):
        # User shouldn't be able to publish a blogpost
        res = tapp.post( f"/blogposts/{test_blogpost.id}/publish")
        assert "not authorised" in res.json["error"]
        assert res.status_code == 401
```

My fixtures live inside a `/tests/fixtures` module.
```
tests/
├─ conftest.py
├─ test_blogs.py
├─ fixtures/
│  ├─ blogs.py

```

My first thought was to manually import fixtures in my test_ files so that I can easily find where they are located. Is it a common practice?

I came across this VSCode extension: https://marketplace.visualstudio.com/items?itemName=Cameron.vscode-pytest. but it doesn't seem to work for me. Anyone using it successfully? 

This article [[Modularizing Pytest Fixtures](https://medium.com/@nicolaikozel/modularizing-pytest-fixtures-fd40315c5a93)] and this SO answer [[In pytest, what is the use of conftest.py files?](https://stackoverflow.com/a/34520971)] were quite helpful.

**Comment by jonathanmach** [comment]
In large projects with hundreds of fixtures, how to easily navigate to the file where  each fixture lives? Something link Ctrl+click to jump to the fixture definition...
I want our developers to be able to quickly inspect our fixtures and learn about other fixtures existing in the same file

**Reply by symonk** [reply]
Will have a look into VSCode when I am home later, but this works no problems on `pycharm`, can `CTRL+click` through to the definition, I'd assume VSCode would of been the same

**Reply by jonathanmach** [reply]
Oh, great to know that it works fine on Pycharm! There should be a VScode specific setting for this, or an extension

**Comment by tabeatheunicorn** [comment, marked as answer]
Maybe a little late, but this was the first result in my Google search:
VSCode added that feature https://code.visualstudio.com/updates/v1_76#_improved-intellisense-support-for-pytest as discussed here https://stackoverflow.com/questions/67601759/support-intellisense-for-pytest-fixtures-in-vscode

**Reply by fohrloop** [reply]
Just to add to this. To enable, add this to your settings JSON:

```json
  "python.analysis.inlayHints.pytestParameters": true,
```

**Comment by chalasanim** [comment]
Please run   '​​pytest​​ ​​--fixtures​​ ​​-v​' 

The fixtures found in conftest.py files are at the bottom. If you supply a directory, pytest will list the fixtures available to tests in that directory. If you supply a test file name, pytest will include those defined in test modules as well.
